src/main/java/seedu/apollo/ui/Parser.java

```diff
@@ -136,11 +146,18 @@ private static Command parseCommand(String[] split, int size, ModuleList moduleD
             return new ExitCommand();
 
         case COMMAND_HELP_WORD:
-            if (!isOneWord(split)) {
-                throw new IllegalCommandException();
+            if (isOneWord(split)){
+
+                return new HelpCommand();
+            }
+            if (isTwoWord(split)) {
+                return chooseHelpCommand(split[1]);
+
             }
-            return new HelpCommand();
 
+            else if(split.length > 2) {
```

Comment:
You should move this above the 2 checks so it will immediate filter out everything with more than 2 words